### ISSUE TITLE
Can't detect headphones whose name has been changed 

### ISSUE THREAD

**Issue by byf3332** [TheLastGimbus/FreeBuddy#6]
The headphones' name can be changed via setting on a HUAWEI phone. After the change Freebuddy cannot detect the headphones.
Maybe you can try detect through MAC address that belongs to HUAWEI.
System:Havoc OS on Android Q 
![Screenshot_20230524-095105](https://github.com/TheLastGimbus/FreeBuddy/assets/71202267/c02f1877-0284-480b-9759-54b5359874d1)
![Screenshot_20230524-095111](https://github.com/TheLastGimbus/FreeBuddy/assets/71202267/6c539a93-9b8f-4719-acae-c7f1b2e6afb5)

**Comment by TheLastGimbus** [comment]
Ah damn. On my LineageOS, this changes the `device.alias` property, and I even use it in the app as their name!

Your mac is `10:5D:DC:8D:E8:1F`, right? i'll try to find matching parts with mine...

**Comment by byf3332** [comment]
> Your mac is `10:5D:DC:8D:E8:1F`, right?
Yes, that's my headphones' mac address 
>  System:Havoc OS
> Wait, so you have huawei or havoc custom rom?
No, this is another device.
(However my huawei phone is also unlocked and rooted on emui 9 though, geuss i'll try a android 10 rom on it later, as i found a custom kernel recently)

**Comment by TheLastGimbus** [comment]
So, when you changed their name on huawei device, they changed it permanently?? Interesting... maybe there's some hidden api for that that we could use too :grin:

**Comment by byf3332** [comment]
> So, when you changed their name on huawei device, they changed it permanently??

Yes, requires specified system version of huawei device (requires EMUI 10 or up ?, I changed the name on emui 10. Also I can confirm that on Harmony OS 2 it works ), and the name change is permanent, you can see it on another device just like Apple Airpods.
However now I don't have emui 10 system so I'm unable to test it now (To root I have to downgrade to emui 9 to unlock bootloader).

**Comment by byf3332** [comment]
Can you tell me where is the code for detecting headphones? I 'll try to fork and build one using my name 🤔
I barely know anything about flutter, but thanks to your actions I can build apks for me.

**Comment by byf3332** [comment]
Found a temporary solution for rooted device:
Go to`/data/misc/bluedroid/`,there will be 2 files
`bt_config.conf`and`bt_config.bak`,which comtains the same thing🤔.
Turn bluetooth off, and use a text editor to open both of the file, change the `Name = xxx`to`Name = HUAWEI FreeBuds 4i` which is under your device's MAC in both file, then save. Now You shall see The name in bluetooth devices list has changed to what you set in the file, and Freebuddy should work now. 
![Screenshot_20230603-204609](https://github.com/TheLastGimbus/FreeBuddy/assets/71202267/fb9c418a-fa56-4557-bafc-489a149e10f2)
![Screenshot_20230603-204557](https://github.com/TheLastGimbus/FreeBuddy/assets/71202267/9603c9ae-b8a6-4b2d-82ca-a364136cdc2a)

**Comment by TheLastGimbus** [comment]
In general, i will need to write some nicer algo that would read UUID advertisiments - tho this will require some messing with bluetooth apis (nothing pleasant :vomiting_face: )
